docs(interfaces): document important interfaces and classes

staltz opened the pull request:
Write JSDoc documentation for Observer, Subscriber, Subscription, Notification, Scheduler, Action,
etc.

blesh commented on src/Observer.ts:
```diff
@@ -21,6 +21,68 @@ export interface CompletionObserver<T> {
+ * @name Observer
+ * @noimport true
+ */
+export class ObserverDoc<T> {
```
interesting, does this build as part of the output? That wouldn't be ideal.

Too bad there isn't a way to document interfaces. Perhaps we add some files to the build that are purely for documenting interfaces and build them separately?

staltz replied:
Yeah it's ugly, I know.

> Perhaps we add some files to the build that are purely for documenting interfaces and build them separately?

We can do that. The real reason is that TypeScript strips away JSDoc comments for interfaces. And I won't try another PR to TS in vain if they are so... 'courteous'.

kwonoj replied:
Agree it's pain to deal with comment strip in typescript with docs. :/

staltz commented:
I rebased and also moved interface documentations to files `MiscJSDoc.ts` which are not compiled to any release bundle, but appear in the ESDoc site.

Can haz merge now?

![cuteness](https://cloud.githubusercontent.com/assets/90512/14139293/d5ac825c-f67a-11e5-9716-ad743f3c0f77.jpg)

blesh commented:
LGTM.

staltz commented:
Rebased. :white_check_mark:

kwonoj commented:
Merged with 8da7a8f, thanks @staltz :)
